File: frontend/src/types/sensei.ts
export type SenseiConceptItem = {
  title: string;
  definition: string;
  key_points: string[];
  example: string;
  code_example?: string;
};

export type SenseiPracticeQuestion = {
  question: string;
  answer: string;
};

export type SenseiContentResponse = {
  topic: string;
  concepts: SenseiConceptItem[];
  practice_questions: SenseiPracticeQuestion[];
};

export type SenseiChatMessage = {
  role: "user" | "assistant";
  content: string;
};

export type ChatHistoryMessage = {
  id: number;
  role: "user" | "assistant";
  content: string;
};

export type ChatHistoryResponse = {
  task_id: number;
  messages: ChatHistoryMessage[];
};

File: backend/app/schemas/sensei.py
from pydantic import BaseModel
from typing import List, Optional


class ConceptItem(BaseModel):
    title: str
    definition: str
    key_points: List[str]
    example: Optional[str] = None
    code_example: Optional[str] = None


class PracticeQuestion(BaseModel):
    question: str
    answer: str


#Content

class SenseiContentRequest(BaseModel):
    topic: str
    course_name: str
    course_id: int


class SenseiContentResponse(BaseModel):
    topic: str
    concepts: List[ConceptItem]
    practice_questions: List[PracticeQuestion]


#Chat

class ChatMessage(BaseModel):
    role: str       # "user" | "assistant"
    content: str


class SenseiChatRequest(BaseModel):
    topic: str
    course_name: str
    history: List[ChatMessage]
    message: str


class SenseiChatResponse(BaseModel):
    reply: str


# Chat History

class ChatHistoryMessage(BaseModel):
    id: int
    role: str
    content: str

class ChatHistoryResponse(BaseModel):
    task_id: int
    messages: List[ChatHistoryMessage]

class SaveMessageRequest(BaseModel):
    task_id: int
    role: str
    content: str
